stdlib/src/builtin/string_literal.mojo

```diff
@@ -530,28 +530,28 @@ struct StringLiteral(
 
     @always_inline
     fn as_bytes(self) -> Span[Byte, StaticConstantOrigin]:
-        """
-        Returns a contiguous Span of the bytes owned by this string.
+        """Returns a contiguous slice of bytes.
 
         Returns:
-            A contiguous slice pointing to the bytes owned by this string.
-        """
+            A contiguous slice pointing to bytes.
 
+        Notes:
+            This does not include the trailing null terminator.
+        """
         return Span[Byte, StaticConstantOrigin](
             ptr=self.unsafe_ptr(), length=self.byte_length()
         )
 
     @always_inline
     fn as_bytes(ref self) -> Span[Byte, __origin_of(self)]:
-        """Returns a contiguous slice of the bytes owned by this string.
+        """Returns a contiguous slice of bytes.
 
         Returns:
-            A contiguous slice pointing to the bytes owned by this string.
+            A contiguous slice pointing to bytes.
```

Comment:
This seems grammatically/contextually confusing. If `bytes` was a variable it would make sense, but in this case it doesn't read quite right.

Reply:
bytes is the plural of a byte (?), and a contiguous slice of bytes is `[byte, byte, ...]` (?). Not sure how that's confusing. I'll change it to "utf-8 bytes" if that solves it

Reply:
Nvm, I think it will be more restraining on the API to say explicitly UTF-8, these bytes could encode anything. [Bytes is a type in Python](https://docs.python.org/3/library/stdtypes.html#bytes-and-bytearray-operations), so this makes sense to me.

Reply:
I could change just the `String`, `StringLiteral`, and `StringSlice` docs to say UTF-8 bytes though WDYT?.

Reply:
My point was more:

>  A contiguous slice pointing to bytes.

What bytes? Who owns them? What do they contain? While this may seem obvious, your change removes that context.

Reply:
> What bytes? [...] What do they contain?

UTF-8 bytes

> Who owns them?

The origin that the Span carries

---
My main goal is to eventually abstract this API over all Stringlike types, owning and non-owning, once the compiler Origin issues are resolved (see PR [#3677 files](https://github.com/modularml/mojo/pull/3677/files#diff-a4b7a6c75bf9e4a8328eee2f7d6109fd2310cc26c905e24f39c3db572ae1aaa0R487))

Reply:
I'm still not seeing how this is more descriptive as a documentation line. It seems less descriptive to me.

Reply:
Also I'd hold off a little on string unification work until the team has time to discuss your proposal. We should definitely improve the general API surface, but we need to think as a team how we want to handle it. For example, implicit conversion into `StringSlice` will handle a _lot_ of the general string cases.

Reply:
Because I'm trying to make the documentation consistent across all `Stringlike` types, not just owning ones. And that means making it more generic. I can add the encoding (`UTF-8`), but even that information might not hold in the future if we implement `UTF-16` & `UTF-32` as a parametrized `String`, and since Python uses `encode(string, encoding) -> bytes` represents all possible encodings, that complicates things even further.

Reply:
I agree with @lsh that removing "the bytes owned by this string" isn't an improvement.   You haven't made a compelling case for why having less information here helps people using the type. 

There's no requirement that different types implementing the same trait have identical API doc for its methods. And where those methods have differences—like how the data is owned—the API doc should note it.